### ISSUE TITLE
Apparent bug in Samsung's "Internet" Browser's custom tab implementation

### ISSUE THREAD

**Issue by ddccmdev** [openid/AppAuth-Android#116]
My company's use of AppAuth hit a snag when we couldn't login on our Galaxy S7 and Galaxy S7 Edge test devices in their factory-default setting which is using Samsung's "Internet" as the default browser.  This browser apparently is also a Custom Tabs provider, and it appears that when using Samsung's Custom Tabs implementation it eats the redirect URIs.

What I mean is this: our Android app registers as the handler for a particular URI, and our login server redirects to that URI at the end of the login process -- but Android never gets that redirect and control never returns to our app once login is complete.  When we switch to using Chrome Custom Tabs, everything works as expected.

We're currently experimenting with our URIs to see if it only eats certain URI pattern, and we'll also be filing an issue with Samsung.  We've currently worked around the issue by making a local fork of AppAuth and hard-coding a "Use Chrome's Custom Tabs" rule.

Have you considered allowing AppAuth users to have more control over which Custom Tabs implementations the do (or do not) want to use?

**Comment by mattinger** [comment]
We have hit this too, and of particular note is the log output.  This looks very much like the recent bug in chrome 53 that i report in #114 

`10-14 10:27:19.481 3064-3708/? I/audio_hw_primary: do_out_standby +
10-14 10:27:19.531 3064-3708/? D/audio_hw_primary: output_devices devices:0
10-14 10:27:19.531 3064-3708/? D/audio_hw_primary: output_devices devices:0
10-14 10:27:19.531 3064-3708/? V/audio_hw_primary: select_devices output_scenario:-1 input_scenario:-1 out_snd_device 0x0 in_snd_device:0x0
10-14 10:27:19.541 3064-3708/? I/audio_hw_primary: set_spk_rx_mute mute(1), rx val=0
10-14 10:27:19.541 3064-3708/? I/audio_route: > audio_route_reset :
10-14 10:27:19.541 3064-3708/? I/audio_route: > audio_route_update_mixer : +
10-14 10:27:19.541 3064-3708/? I/audio_route: > audio_route_update_mixer : changed(7) -
10-14 10:27:19.541 3064-3708/? I/audio_hw_primary: select_devices - 
10-14 10:27:19.541 3064-3708/? I/audio_hw_primary: do_out_standby -
10-14 10:27:20.671 3490-3946/? D/InputReader: Input event(4): value=1 when=2029609388000
10-14 10:27:20.671 3490-3946/? D/InputReader: Input event(4): value=1 when=2029609388000
10-14 10:27:20.671 3490-3946/? I/InputReader: Touch event's action is 0x0 (deviceType=0) [pCnt=1, s=0.484 ] when=2029609388000
10-14 10:27:20.671 3490-3946/? D/InputReader: lastThreadEndTime = 2027631250841, currentThreadStartTime = 2027631258091, diff = 0
10-14 10:27:20.671 3490-3945/? I/InputDispatcher: Delivering touch to (4065): action: 0x4, toolType: 1
10-14 10:27:20.671 3490-3945/? I/InputDispatcher: Delivering touch to (22216): action: 0x0, toolType: 1
10-14 10:27:20.671 22216-22216/? D/ViewRootImpl: ViewPostImeInputStage processPointer 0
10-14 10:27:20.671 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2413)] WebViewImpl.cpp :: handleInputEvent :: event received = 19
10-14 10:27:20.741 3490-3946/? D/InputReader: Input event(4): value=0 when=2029688270000
10-14 10:27:20.741 3490-3946/? D/InputReader: Input event(4): value=0 when=2029688270000
10-14 10:27:20.741 3490-3946/? I/InputReader: Touch event's action is 0x1 (deviceType=0) [pCnt=1, s=] when=2029688270000
10-14 10:27:20.741 3490-3945/? I/InputDispatcher: Delivering touch to (22216): action: 0x1, toolType: 1
10-14 10:27:20.741 22216-22216/? D/ViewRootImpl: ViewPostImeInputStage processPointer 1
10-14 10:27:20.741 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2413)] WebViewImpl.cpp :: handleInputEvent :: event received = 18
10-14 10:27:20.921 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2413)] WebViewImpl.cpp :: handleInputEvent :: event received = 16
10-14 10:27:20.921 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2413)] WebViewImpl.cpp :: handleInputEvent :: event received = 17
10-14 10:27:20.931 6317-6317/? D/SamsungIME: Dismiss ExpandCandiate
10-14 10:27:21.081 3490-4675/? E/BoostFramework: Exception java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[])' on a null object reference
                                                 
                                                 --------- beginning of system
10-14 10:27:21.081 3490-3490/? D/LightsService: [api] [SvcLED] turnOff:: id = 6 (uid: 1000 pid: 3490) 
10-14 10:27:21.081 3490-3490/? D/LightsService: [SvcLED] setSvcLedStateLocked:: SvcLEDState : 0x0 -> 0x0 | SvcLED(id=6) set Off
10-14 10:27:21.081 3490-3490/? D/EdgeLight: Turning on edge light for notification (NotificationRecord(0x02f4b2e7: pkg=com.google.android.gm user=UserHandle{0} id=643823870 tag=null score=0 key=0|com.google.android.gm|643823870|null|10118: Notification(pri=0 contentView=null vibrate=null sound=content://settings/system/notification_sound tick defaults=0x4 flags=0x19 color=0xffdb4437 category=email actions=2 vis=PRIVATE publicVersion=Notification(pri=0 contentView=com.google.android.gm/0x10900a3 vibrate=null sound=null defaults=0x0 flags=0x0 color=0xffdb4437 category=email vis=PUBLIC secFlags=0x0 secPriority=0) secFlags=0x0 secPriority=0))) with color=ffffffff
10-14 10:27:21.081 6317-6317/? E/KeyboardInfoUtils: getInstance start
10-14 10:27:21.081 6317-6317/? E/KeyboardInfoUtils: sendSIPInformation state:6  isAbstractKeyboardView : true
10-14 10:27:21.091 3490-3608/? D/LightsService: [SvcLED] handleForcedSvcLEDTask()
10-14 10:27:21.081 6317-6317/? E/KeyboardInfoUtils: sending null keyboardInfo as SIP is closed
10-14 10:27:21.091 3490-3608/? D/LightsService: [SvcLED]  setSvcLedLightLocked :: Current SvcLED(id=6) maintains its priority right
10-14 10:27:21.091 6317-6317/? W/ContextImpl: Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:830 android.content.ContextWrapper.sendBroadcast:410 com.sec.android.inputmethod.base.view.KeyboardInfoUtils.sendSIPInformation:503 com.sec.android.inputmethod.base.common.InputManagerImpl.sendSIPClosedBroadcastToUniversalSwitch:6745 com.sec.android.inputmethod.base.common.InputManagerImpl.finishInputView:6728 
10-14 10:27:21.091 3490-3490/? D/MARsPolicyManager: NotificationListenerService OngoingNotificationRemoved : android
10-14 10:27:21.091 6317-6317/? D/SamsungIME: onWindowHidden
10-14 10:27:21.091 6317-6317/? D/SamsungIME: notifyWindowStateChanged  Window Hidden. AutoText consider to shutdown
10-14 10:27:21.091 6317-6317/? W/ContextImpl: Calling a method in the system process without a qualified user: android.app.ContextImpl.sendBroadcast:830 android.content.ContextWrapper.sendBroadcast:410 android.inputmethodservice.InputMethodService.sendInputViewShownState:443 android.inputmethodservice.InputMethodService.hideWindow:2310 android.inputmethodservice.InputMethodService.doHideWindow:2288 
10-14 10:27:21.091 4065-4065/? D/PhoneStatusBar: removeNotification key=-1|android|17040374|InputMethodManagerService_Noti|1000 old=StatusBarNotification(pkg=android user=UserHandle{-1} id=17040374 tag=InputMethodManagerService_Noti score=-20 key=-1|android|17040374|InputMethodManagerService_Noti|1000: Notification(pri=-2 contentView=android/0x10900a3 vibrate=null sound=null defaults=0x0 flags=0x1002 color=0x00000000 category=sys vis=PRIVATE secFlags=0x0 secPriority=0))
10-14 10:27:21.091 4065-4065/? D/PhoneStatusBar: setAreThereNotifications: N=5 any=true clearable=true
10-14 10:27:21.101 3490-3490/? D/InputMethodManagerService: [HARDWARE_KEYBOARD] (refreshImeWindowVis) mImeWindowVis= 0 inputVisible = falsehaveHardKeyboard = false hardKeyShown = false
10-14 10:27:21.111 3044-4737/? D/libEGL: eglTerminate EGLDisplay = 0x7f8819de78
10-14 10:27:21.111 3044-4737/? D/libEGL: eglTerminate EGLDisplay = 0x7f8819de78
10-14 10:27:21.121 22216-22216/? D/ViewRootImpl: MSG_RESIZED: ci=Rect(0, 96 - 0, 0) vi=Rect(0, 96 - 0, 0) or=1
10-14 10:27:21.131 22216-22216/? D/KeyboardUtils: isMobileKeyboardConnected : isCovered/constYes:0/1
10-14 10:27:21.141 22216-22216/? D/KeyboardUtils: isMobileKeyboardConnected : isCovered/constYes:0/1
10-14 10:27:21.141 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(1993)] WebViewImpl::resize->START :: newSize = 360,566:old size = 360,284
10-14 10:27:21.141 22729-22743/? I/chromium: [INFO:FrameView.cpp(955)] FrameView::layout
10-14 10:27:21.141 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2022)] WebViewImpl::resize :: CONTENT SIZE after LAYOUT :: 360,566
10-14 10:27:21.141 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2023)] WebViewImpl::resize :: LAYOUT SIZE after LAYOUT :: 360,566
10-14 10:27:21.141 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2024)] WebViewImpl::resize->END
10-14 10:27:21.151 6317-6317/? D/SamsungIME: SwiftkeyWrapper-checkCurrentSequence : use IC
10-14 10:27:21.171 22729-22743/? I/chromium: [INFO:CompositedDeprecatedPaintLayerMapping.cpp(2201)] CompositedDeprecatedPaintLayerMapping::paintContents
10-14 10:27:21.251 3490-6897/? D/SSRM:n: SIOP:: AP = 380, PST = 384 (W:14), CP = 300, CUR = 49, LCD = 104
10-14 10:27:21.461 3044-4737/? I/SurfaceFlinger: id=289 Removed JnputMethod (8/9)
10-14 10:27:21.461 3044-3142/? I/SurfaceFlinger: id=289 Removed JnputMethod (-2/9)
10-14 10:27:21.471 3044-3044/? D/libEGL: eglTerminate EGLDisplay = 0x7fd2c9a9d8
10-14 10:27:24.501 3072-3132/? D/Sensorhubs: readContextData: 1, 1, 7, -1
10-14 10:27:24.501 3490-3936/? D/SensorHubManager: onGetSensorHubDataLocked: library(4) = 1, 1, 7, -1
10-14 10:27:24.501 3490-3935/? D/CAE: onGetSensorHubData(SensorHubParserProvider.java:92) - onGetSensorHubData Event [event buffer len :4], AP_WAKEUP
10-14 10:27:24.501 3490-3935/? I/CAE: parse(SensorHubParserProvider.java:188) - buffer size = 4
10-14 10:27:24.501 3490-3935/? I/CAE: parse(SensorHubParserProvider.java:199) - 1, 1, 7, -1,
10-14 10:27:24.501 3490-3935/? D/CAE: display(ContextProvider.java:375) - ================= AUTO_ROTATION =================
10-14 10:27:24.501 3490-3935/? I/CAE: display(ContextProvider.java:391) - Angle=[-1]
10-14 10:27:24.501 3490-3938/? D/SContextService: updateSContext() : event = Auto Rotation
10-14 10:27:24.501 3490-3490/? V/WindowOrientationListener: OrientationSensorJudge.onSContextChanged, Rotation: -1
10-14 10:27:25.471 3490-4738/? D/BatteryService: !@BatteryListener : batteryPropertiesChanged!
10-14 10:27:25.471 3490-4738/? D/BatteryService: level:46, scale:100, status:2, health:2, present:true, voltage: 3843, temperature: 313, technology: Li-ion, AC powered:false, USB powered:true, POGO powered:false, Wireless powered:false, icon:17303412, invalid charger:0, maxChargingCurrent:0
10-14 10:27:25.471 3490-4738/? D/BatteryService: online:4, current avg:162, charge type:1, power sharing:false, high voltage charger:false, capacity:280000, batterySWSelfDischarging:false, current_now:268
10-14 10:27:25.471 3490-3490/? D/BatteryService: Sending ACTION_BATTERY_CHANGED.
10-14 10:27:25.481 3490-3490/? I/MotionRecognitionService: Plugged
10-14 10:27:25.481 3490-3490/? E/MotionRecognitionService: support TA ~
10-14 10:27:25.481 3490-3490/? I/MotionRecognitionService: isTAConnected | current backoffstate  = 33792
10-14 10:27:25.481 3490-3490/? D/MotionRecognitionService:  TA connected ,  33792
10-14 10:27:25.481 3490-3490/? I/MotionRecognitionService: skip setTransmitPower. 
10-14 10:27:25.481 3490-3490/? D/MotionRecognitionService:   cableConnection= 1
10-14 10:27:25.481 3490-3490/? D/MotionRecognitionService: setPowerConnected | current backoffstate  = 33792 , state =33792
10-14 10:27:25.481 3490-3490/? D/MotionRecognitionService: skip setTransmitPower. 
10-14 10:27:25.491 4065-4065/? D/KeyguardUpdateMonitor: received broadcast android.intent.action.BATTERY_CHANGED
10-14 10:27:25.491 4065-4065/? D/KeyguardUpdateMonitor: handleBatteryUpdate
10-14 10:27:25.491 4065-4065/? D/PowerUI: priorPlugType = 2 mPlugType =  2
10-14 10:27:25.491 4065-4065/? D/PowerUI.Notification: showChargingNotice oldChargingType : 1 currentChargingType : 1 oldChargingTime : 11891000 mChargingTime : 11891000
10-14 10:27:25.491 4065-4065/? D/PowerUI.Notification: There is no change about charging status, so return!
10-14 10:27:25.501 4065-4065/? D/BatteryMeterView: ACTION_BATTERY_CHANGED : level:46 status:2 health:2
10-14 10:27:25.501 5561-5561/? D/CommonServiceConfiguration: getStringValueSetting
10-14 10:27:25.511 5561-5561/? D/BatteryMonitor: new battery level: 46
10-14 10:27:25.531 4065-4065/? D/BatteryMeterView: ACTION_BATTERY_CHANGED : level:46 status:2 health:2
10-14 10:27:28.851 3490-3946/? D/InputReader: Input event(4): value=1 when=2037783501000
10-14 10:27:28.851 3490-3946/? D/InputReader: Input event(4): value=1 when=2037783501000
10-14 10:27:28.851 3490-3946/? I/InputReader: Touch event's action is 0x0 (deviceType=0) [pCnt=1, s=0.485 ] when=2037783501000
10-14 10:27:28.851 3490-3946/? D/InputReader: lastThreadEndTime = 2029688452925, currentThreadStartTime = 2029688454759, diff = 0
10-14 10:27:28.851 3490-3945/? I/InputDispatcher: Delivering touch to (4065): action: 0x4, toolType: 1
10-14 10:27:28.851 3490-3945/? I/InputDispatcher: Delivering touch to (22216): action: 0x0, toolType: 1
10-14 10:27:28.851 22216-22216/? D/ViewRootImpl: ViewPostImeInputStage processPointer 0
10-14 10:27:28.851 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2413)] WebViewImpl.cpp :: handleInputEvent :: event received = 19
10-14 10:27:28.951 3490-3946/? D/InputReader: Input event(4): value=0 when=2037896709000
10-14 10:27:28.951 3490-3946/? D/InputReader: Input event(4): value=0 when=2037896709000
10-14 10:27:28.951 3490-3946/? I/InputReader: Touch event's action is 0x1 (deviceType=0) [pCnt=1, s=] when=2037896709000
10-14 10:27:28.951 3490-3945/? I/InputDispatcher: Delivering touch to (22216): action: 0x1, toolType: 1
10-14 10:27:28.951 22216-22216/? D/ViewRootImpl: ViewPostImeInputStage processPointer 1
10-14 10:27:28.951 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2413)] WebViewImpl.cpp :: handleInputEvent :: event received = 18
10-14 10:27:29.091 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2413)] WebViewImpl.cpp :: handleInputEvent :: event received = 16
10-14 10:27:29.091 22729-22743/? I/chromium: [INFO:WebViewImpl.cpp(2413)] WebViewImpl.cpp :: handleInputEvent :: event received = 17
10-14 10:27:29.141 3490-3506/? D/SecContentProvider: query(), uri = 10 selection = isUrlBlocked
10-14 10:27:29.141 3490-4727/? D/SecContentProvider: query(), uri = 2 selection = isAuditLogEnabled
10-14 10:27:29.401 3490-4738/? D/SecContentProvider: query(), uri = 10 selection = isUrlBlocked
10-14 10:27:29.401 3490-5742/? D/SecContentProvider: query(), uri = 2 selection = isAuditLogEnabled
10-14 10:27:30.631 3490-3933/? V/AlarmManager: Expired Alarm result :8
10-14 10:27:30.651 3490-3562/? V/BroadcastQueue: [background] Process cur broadcast BroadcastRecord{ef7d6f9 u0 null qIdx=4}, state= (APP_RECEIVE) DELIVERED for app ProcessRecord{7025920 5276:com.google.android.gms.persistent/u0a11}
10-14 10:27:30.721 3490-3969/? I/ActivityManager: Start proc 23440:com.sec.android.app.sbrowser:sandboxed_process3/u0i12 for service com.sec.android.app.sbrowser/org.chromium.content.app.SandboxedProcessService3
10-14 10:27:30.721 23440-23440/? E/Zygote: v2
10-14 10:27:30.721 23440-23440/? I/libpersona: KNOX_SDCARD checking this for 99012
10-14 10:27:30.731 23440-23440/? I/libpersona: KNOX_SDCARD not a persona
10-14 10:27:30.731 23440-23440/? E/Zygote: isSdpEnabledProcess - SDP enabled
10-14 10:27:30.731 23440-23440/? I/SELinux: Function: selinux_compare_spd_ram, index[1], SPD-policy is existed. and_ver=SEPF_SECMOBILE_6.0.1 ver=11
10-14 10:27:30.731 23440-23440/? W/SELinux: Function: selinux_compare_spd_ram, index[1], priority [1], priority version is VE=SEPF_SECMOBILE_6.0.1_0011
10-14 10:27:30.731 23440-23440/? E/Zygote: accessInfo : 64
10-14 10:27:30.731 23440-23440/? E/Zygote: isSdpEnabledProcess - SDP enabled
10-14 10:27:30.731 23440-23440/? E/Zygote: setSDPgroupsIntarray[pid]: 0 / [gid]: 99012 / [uid]: 99012
10-14 10:27:30.731 23440-23440/? W/SELinux: SELinux: seapp_context_lookup: seinfo=platform, level=s0:c512,c768, pkgname=com.sec.android.app.sbrowser:sandboxed_process3 
10-14 10:27:30.751 23440-23440/? D/TimaKeyStoreProvider: TimaSignature is unavailable
10-14 10:27:30.751 23440-23440/? D/ActivityThread: Added TimaKeyStore provider
10-14 10:27:30.771 5276-23437/? D/SecWifiDisplayUtil: Metadata value : SecSettings2
10-14 10:27:30.781 23440-23440/? W/ResourcesManager: getTopLevelResources: /data/app/com.sec.android.app.sbrowser-1/base.apk / 1.0 running in com.sec.android.app.sbrowser rsrc of package com.sec.android.app.sbrowser
10-14 10:27:30.781 23440-23440/? D/ResourcesManager: For user 0 new overlays fetched Null
10-14 10:27:30.791 23440-23440/? I/InjectionManager: Inside getClassLibPath caller 
10-14 10:27:30.811 23440-23440/? D/ApplicationStatus: initialize application (com.sec.android.app.sbrowser.common.SBrowserMobileApplication@2b345b0)
10-14 10:27:30.831 23440-23440/? D/InjectionManager: InjectionManager
10-14 10:27:30.831 23440-23440/? I/InjectionManager: featureStore :{}
10-14 10:27:30.831 23440-23440/? I/ChildProcessService: Creating new ChildProcessService pid=23440
10-14 10:27:30.871 23440-23453/? W/LibraryLoader: Loadinng a native library: chromium_android_linker
10-14 10:27:30.871 23440-23453/? W/LibraryLoader: Loadinng a native library: v8
10-14 10:27:30.881 23440-23453/? W/LibraryLoader: Loadinng a native library: swebview
10-14 10:27:30.921 23440-23453/? I/LibraryLoader: Time to load native libraries: 48 ms (timestamps 9813-9861)
10-14 10:27:30.921 23440-23453/? I/LibraryLoader: Expected native library version number "44.0.2403.133",actual native library version number "44.0.2403.133"
10-14 10:27:30.921 23440-23453/? I/chromium: [INFO:library_loader_hooks.cc(134)] Chromium logging enabled: level = 0, default verbosity = 0
10-14 10:27:30.921 23440-23453/? I/ChildProcessService: onCreate - Native Initialization start
10-14 10:27:30.921 23440-23453/? W/ResourcesManager: getTopLevelResources: /data/app/com.sec.android.app.sbrowser-1/base.apk / 1.0 running in com.sec.android.app.sbrowser rsrc of package com.sec.android.app.sbrowser
10-14 10:27:30.931 23440-23453/? E/chromium: [ERROR:process_sighandler.cc(129)] Enabled sigaction handler.
10-14 10:27:30.941 23440-23453/? E/ChildProcessService: onCreate - Native Initialization failed
10-14 10:27:30.941 23440-23453/? I/chromium: [INFO:child_process_service.cc(179)] ChildProcessService: Exiting child process.
10-14 10:27:30.961 22216-22216/? W/ChildProcessConnection: onServiceDisconnected (crash or killed by oom): pid=23440
10-14 10:27:30.961 3490-4727/? I/ActivityManager: Process com.sec.android.app.sbrowser:sandboxed_process3 (pid 23440)(adj 1) has died(58,1016)
10-14 10:27:30.961 3490-3556/? W/libprocessgroup: failed to open /acct/uid_10151/pid_23440/cgroup.procs: No such file or directory
10-14 10:27:30.961 3490-4727/? D/ActivityManager: isAutoRunBlockedApp:: com.sec.android.app.sbrowser, Auto Run ON
10-14 10:27:30.961 3490-4727/? W/ActivityManager: Scheduling restart of crashed service com.sec.android.app.sbrowser/org.chromium.content.app.SandboxedProcessService3 in 1000ms
10-14 10:27:30.971 22216-22216/? I/chromium: [INFO:render_process_host_impl.cc(1952)] [Renderer][Exit] ChildProcessLauncher::OnChildProcessDied called through JNI pid = 23440, renderer_id =0
10-14 10:27:31.281 3490-6897/? D/SSRM:n: SIOP:: AP = 360, PST = 383 (W:18), CP = 299, CUR = 162, LCD = 104`

**Comment by prodaea** [comment]
By using a blacklist or whitelist setup clients would have to catch an exception if the desired browser isn't installed or enabled and produce messaging for the user that login can only be completed by an arbitrary browser.

But we're also seeing this behavior and also hardcoded to use chrome for custom tabs, but default to whichever browser is available - without custom tabs. Our issue was then that other browsers like firefox start in a new task. The result was that when firefox returned to the AuthenticationManagementActivity it had no state and would crash. I have fixed this in our app by adding `launchMode="singleTask"` to the AMA's manifest. I think this is sub-optimal, but it's best effort at this point.

**Comment by mattinger** [comment]
PS:  I also think it's worth noting that the issues related to AuthenticationManagementActivity do not exist on the released version of 0.3.0.  The AuthenticationManagementActivity was introduced post 0.3.0 (ironically, i believe to deal with an issue i created related to knowing when the user closed the browser tab manually).

**Comment by iainmcgin** [comment]
Before I do a 0.4.0 release I'll test the non-custom tab case and see what the right approach is to fixing any issues there. #119 should at least give apps the ability to control which browser is used, to avoid known broken versions of browsers.

**Comment by mattinger** [comment]
I'm also noticed that the crash is very similar to the chrome 53 crash, which makes sense, as it seems that samsung's browser is based on chromium, so this may resolve itself with a new version of that browser.

**Comment by iainmcgin** [comment]
Blacklisting of misbehaving browsers and custom tab usage is now available in HEAD and will be included in the 0.4.0 release.

**Comment by mattinger** [comment]
Is there a way to talk to Samsung about this?  (I have no contacts there).  A large # of phones might only have the samsung stock browser installed, and it would be nice if we can get them to fix the issue.

**Comment by ddccmdev** [comment]
I am escalating this with Samsung.

**Comment by iainmcgin** [comment]
Reopening this for visibility - it appears that this issue may still be happening in SBrowser, or older Samsung devices may be pinned to a version of SBrowser that didn't get the fix. Can those with a Samsung device test this and report back?

**Comment by Maritt** [comment]
Is there an updated status on this?

**Comment by KrynaS** [comment]
I am using samsung galaxy S6 and have this issue, got all updates and seems like there will be no more support for this device. I found workaround that work on all browsers, instead of redirecting to http/https://... I do redirect on intent://...

**Comment by axnd** [comment]
This issue has been resolved by Samsung in Samsung Internet 5.4 in May 2017, according to my tests with an S7 Edge and a coworker's S6.

@KrynaS they have subsequently released Samsung Internet 6.2 on October 30th which, according to their [Release Notes](http://developer.samsung.com/internet/android/releases) (attached) supports "phones running Android 5.0 and above" and they no longer seem to have the limitation that it needs to be a Samsung phone.
![screen shot 2017-11-20 at 14 52 57](https://user-images.githubusercontent.com/4670793/33038415-8eeae6ae-ce02-11e7-9017-c8423e20af65.png)

Otherwise, @iainmcgin, I think this ticket can be closed.

**Comment by agologan** [comment]
Per @axnd closing ticket. Thank you.
